app/api/helpers/db.py

```diff
@@ -46,3 +48,22 @@ def safe_query(self, model, column_name, value, parameter_name):
                              "{}: {} not found".format(model.__name__, value))
     else:
         return record
+
+
+def get_or_create(model, **kwargs):
+    was_created = False
+    instance = db.session.query(model).filter_by(**kwargs).first()
+    if instance:
+        return instance, was_created
+    else:
+        instance = model(**kwargs)
+        db.session.add(instance)
+        db.session.commit()
+        was_created = True
+        return instance, was_created
+
+
+def get_count(q):
+    count_q = q.statement.with_only_columns([func.count()]).order_by(None)
```

Comment:
It just resets any order set in the query, since counting is irreverent of any order so through this way we ensure that we are doing any useless ordering in the data. ( was there before in helpers also )